loqrecovery: `TestHalfOnlineLossOfQuorumRecovery` flaked on cluster ID mismatch

Saw this flake over in https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/9067835?showRootCauses=true&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true:

```
=== RUN   TestHalfOnlineLossOfQuorumRecovery
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/04933cbd7686f6ed78f5a446d70daf3f/logTestHalfOnlineLossOfQuorumRecovery3256706910
    test_log_scope.go:79: use -show-logs to present logs inline
[debug recover make-plan --confirm=y --certs-dir=test_certs --host=127.0.0.1:43471 --plan=/artifacts/tmp/_tmp/04933cbd7686f6ed78f5a446d70daf3f/TestHalfOnlineLossOfQuorumRecovery56388490/recovery-plan.json]
    debug_recover_loss_of_quorum_test.go:423:
          Error Trace:  /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7650/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cli/cli_test_/cli_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/debug_recover_loss_of_quorum_test.go:423
          Error:        "debug recover make-plan --confirm=y --certs-dir=test_certs --host=127.0.0.1:43471 --plan=/artifacts/tmp/_tmp/04933cbd7686f6ed78f5a446d70daf3f/TestHalfOnlineLossOfQuorumRecovery56388490/recovery-plan.json\nERROR: failed to retrieve replica info from cluster: rpc error: code = Unknown desc = initial connection heartbeat failed: grpc: client cluster ID \"94c59982-cda3-4ff1-9081-6c7b9281a4e5\" doesn't match server cluster ID \"a2d6c561-7640-4a57-948f-f55c0f5ca16b\" [code 2/Unknown]\n" does not contain "- node n1"
          Test:         TestHalfOnlineLossOfQuorumRecovery
          Messages:     planner didn't provide correct apply instructions
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/04933cbd7686f6ed78f5a446d70daf3f/logTestHalfOnlineLossOfQuorumRecovery3256706910
--- FAIL: TestHalfOnlineLossOfQuorumRecovery (7.66s)
```

Jira issue: CRDB-25387

--- a/pkg/cli/debug_recover_loss_of_quorum_test.go
+++ b/pkg/cli/debug_recover_loss_of_quorum_test.go
@@ -444,29 +444,31 @@ func TestHalfOnlineLossOfQuorumRecovery(t *testing.T) {
 	// cluster using loss of quorum recovery. To do that, we will terminate
 	// two nodes and run recovery on remaining one. Restarting node should
 	// bring it back to healthy (but underreplicated) state.
+	// Note that we inject reusable listeners into all nodes to prevent tests
+	// running in parallel from taking over ports of stopped nodes and responding
+	// to gateway node with errors.
 	// TODO(oleg): Make test run with 7 nodes to exercise cases where multiple
 	// replicas survive. Current startup and allocator behaviour would make
 	// this test flaky.
+	sa := make(map[int]base.TestServerArgs)
+	for i := 0; i < 3; i++ {
+		sa[i] = base.TestServerArgs{
+			Knobs: base.TestingKnobs{
+				Server: &server.TestingKnobs{
+					StickyEngineRegistry: storeReg,
+				},
+			},
+			Listener: listenerReg.GetOrFail(t, i),
+			StoreSpecs: []base.StoreSpec{
+				{
+					InMemory: true,
+				},
+			},
+		}
+	}
 	tc := testcluster.NewTestCluster(t, 3, base.TestClusterArgs{
 		ReusableListeners: true,
-		ServerArgs: base.TestServerArgs{
-			StoreSpecs: []base.StoreSpec{
-				{InMemory: true},
-			},
-		},
-		ServerArgsPerNode: map[int]base.TestServerArgs{
-			0: {
-				Knobs: base.TestingKnobs{
-					Server: &server.TestingKnobs{
-						StickyEngineRegistry: storeReg,
-					},
-				},
-				Listener: listenerReg.GetOrFail(t, 0),
-				StoreSpecs: []base.StoreSpec{
-					{InMemory: true, StickyInMemoryEngineID: "1"},
-				},
-			},
-		},
+		ServerArgsPerNode: sa,
 	})
 	tc.Start(t)
 	s := sqlutils.MakeSQLRunner(tc.Conns[0])
@@ -529,7 +531,8 @@ func TestHalfOnlineLossOfQuorumRecovery(t *testing.T) {
 	require.NoError(t, err, "failed to run apply plan")
 	// Check that there were at least one mention of replica being promoted.
 	require.Contains(t, out, "updating replica", "no replica updates were recorded")
-	require.Contains(t, out, fmt.Sprintf("Plan staged. To complete recovery restart nodes n%d.", node1ID),
+	require.Contains(t, out,
+		fmt.Sprintf("Plan staged. To complete recovery restart nodes n%d.", node1ID),
 		"apply plan failed to stage on expected nodes")
 
 	// Verify plan is staged on nodes
@@ -557,7 +560,8 @@ func TestHalfOnlineLossOfQuorumRecovery(t *testing.T) {
 	ac, err := tc.GetAdminClient(ctx, t, 0)
 	require.NoError(t, err, "failed to get admin client")
 	testutils.SucceedsSoon(t, func() error {
-		dr, err := ac.DecommissionStatus(ctx, &serverpb.DecommissionStatusRequest{NodeIDs: []roachpb.NodeID{2, 3}})
+		dr, err := ac.DecommissionStatus(ctx,
+			&serverpb.DecommissionStatusRequest{NodeIDs: []roachpb.NodeID{2, 3}})
 		if err != nil {
 			return err
 		}

